src/components/TimeLine/TimeLine.tsx

```diff
@@ -0,0 +1,128 @@
+        super(props);
+
+        this.timeLineItems = Array.from(Array(this.props.range.max - this.props.range.min + 1))
+            .fill(undefined)
+            .map((x, i) => this.props.range.min + i);
+    }
+
+    public getChildContext(): TimeLineContext {
+        return {
+            setNextProject: this.handlerNextProject
+        };
+    }
+
+    public componentDidUpdate() {
+        if (this.state.sliderClassName !== this.sliderDefaultClassName) {
+            setTimeout(() => {
+                this.setState({
+                    sliderClassName: this.sliderDefaultClassName
+                })
+            }, TimeLine.animationDuration * 2);
+        }
+    }
+
+    public render(): JSX.Element {
+        return (
+            <div className="prj-chronology">
+                <this.Slider/>
+                <div className="prj-chronology__div-outer prj-chronology__div-outer_left"/>
+                <this.Container/>
+                <div className="prj-chronology__div-outer prj-chronology__div-outer_right"/>
+            </div>
+        );
+    }
+
+    protected handlerNextProject = (project: ProjectInterface, position: number) => {
+        this.setState({
+            pointPosition: position,
+            sliderClassName: concat(
+                this.sliderDefaultClassName,
+                this.sliderMoveClassName
+            )
+        });
+
+        setTimeout(() => {
+            this.setState({
+                activeProject: project
+            })
+        }, TimeLine.animationDuration);
+    };
+
+    protected Slider = (): JSX.Element => {
+        const sliderOffset = {
+            left: `${this.state.pointPosition}px`
+        };
+
+        return (
+            <div className={this.state.sliderClassName} style={sliderOffset}>
+                <div className="chronology-slider__body"/>
+                <div className="chronology-slider__description">
+                    <div className="chronology-slider__projects">
+                        {this.title}
+                    </div>
+                    <span className="chronology-slider__services">{this.state.activeProject.description}</span>
+                    <span className="chronology-slider__date">{this.date}</span>
+                </div>
+            </div>
+        );
+    };
+
+    protected get date(): string {
+        const activeDate = this.state.activeProject.date;
+        return `${activeDate.year}.${activeDate.month}.${activeDate.day}`;
+    }
+
+    protected get title(): any [] {
+        return this.state.activeProject.title
+            .map(({name, url}, i) => {
+                const props = {
+                    href: url,
+                    key: i,
+                    target: "_blank",
+                    rel: "nofollow noopener"
+                };
+
+                // tslint:disable:jsx-key
+                return i < this.state.activeProject.title.length - 1
+                    ? [<a {...props}>{name}</a>, " / "]
```

Comment:
You should not transform array in this case, just create new with correct key prop and define props using JSX